pallets/gear/src/lib.rs

```diff
@@ -1716,8 +1719,12 @@ pub mod pallet {
             let max_weight = <T as frame_system::Config>::BlockWeights::get().max_block;
             let remaining_weight = max_weight.saturating_sub(weight_used.total());
 
-            // Remaining weight may exceed the minimum block gas limit determined by the Limiter trait
-            let adjusted_gas = GasAllowanceOf::<T>::get().max(remaining_weight.ref_time());
+            // Remaining weight may exceed the minimum block gas limit set by the Limiter trait.
+            let mut adjusted_gas = GasAllowanceOf::<T>::get().max(remaining_weight.ref_time());
+            // Gas for queue processing can never exceed the hard limit, if the latter is provided.
+            if let Some(max_gas) = max_gas {
+                adjusted_gas = adjusted_gas.min(max_gas);
```

Comment:
Here `max_gas` can be only less than the gas allowance. Are there use cases when we want to set `max_gas` greater than the current gas allowance?

Reply:
The target use of this is to limit the queue processing time to some reasonably small value (supposedly smaller than the default block gas limit). If the provided value turns out to be greater than the actual allowance it will simply have no effect on the usual queue processing flow.

Reply:
In my opinion, it would be great to use max gas, e.g., in RPC calls to (say here: https://github.com/gear-tech/gear/blob/master/pallets/gear/src/runtime_api.rs#L110). But if its purpose is only to limit block gas during production, there is no sense in making it bigger indeed.